e2e node test breakage: Heapster canary test failed to build

Jenkins build Node/job/heapster-dockercanarybuild-ci/58/ failed since last Thursday due to compiling issue:

```
04:10:20 Step 5 : RUN cd /go/src/k8s.io/heapster && make && mv heapster /heapster
04:10:20  ---> Running in 53ad68d04c53
04:10:21 rm -f heapster
04:10:21 rm -f eventer
04:10:21 rm -f deploy/docker/heapster
04:10:21 rm -f deploy/docker/eventer
04:10:21 go get github.com/tools/godep
04:10:23 GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go build ./...
04:10:24 # google.golang.org/cloud/compute/metadata
04:10:24 Godeps/_workspace/src/google.golang.org/cloud/compute/metadata/metadata.go:191: req.Cancel undefined (type *http.Request has no field or method Cancel)
04:10:24 godep: go exit status 2
04:10:36 make: *** [build] Error 1
04:10:36 Makefile:14: recipe for target 'build' failed
04:10:37 The command '/bin/sh -c cd /go/src/k8s.io/heapster && make && mv heapster /heapster' returned a non-zero code: 2
04:10:37 Build step 'Docker Build and Publish' marked build as failure
```

@piosz looks like it is caused by https://github.com/kubernetes/heapster/commit/54ecf0e9b4d078b7d901b0cf9edad9e66f5d5f31

Log from `heapster-gce-e2e-ci`:
```
hooks/run_vet.sh
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go build ./...
godep: WARNING: Go version (go1.6) & $GO15VENDOREXPERIMENT= wants to enable the vendor experiment, but disabling because a Godep workspace (Godeps/_workspace) exists
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go build -o heapster k8s.io/heapster/metrics
godep: WARNING: Go version (go1.6) & $GO15VENDOREXPERIMENT= wants to enable the vendor experiment, but disabling because a Godep workspace (Godeps/_workspace) exists
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go build -o eventer k8s.io/heapster/events
godep: WARNING: Go version (go1.6) & $GO15VENDOREXPERIMENT= wants to enable the vendor experiment, but disabling because a Godep workspace (Godeps/_workspace) exists
GOOS=linux GOARCH=amd64 godep go test --test.short -race ./... 
godep: WARNING: Go version (go1.6) & $GO15VENDOREXPERIMENT= wants to enable the vendor experiment, but disabling because a Godep workspace (Godeps/_workspace) exists
ok  	k8s.io/heapster/common/flags	1.021s
?   	k8s.io/heapster/common/gce	[no test files]
?   	k8s.io/heapster/common/influxdb	[no test files]
?   	k8s.io/heapster/common/kubernetes	[no test files]
?   	k8s.io/heapster/events	[no test files]
?   	k8s.io/heapster/events/core	[no test files]
ok  	k8s.io/heapster/events/manager	5.534s
ok  	k8s.io/heapster/events/sinks	16.059s
?   	k8s.io/heapster/events/sinks/gcl	[no test files]
ok  	k8s.io/heapster/events/sinks/influxdb	1.053s
ok  	k8s.io/heapster/events/sinks/log	1.033s
?   	k8s.io/heapster/events/sources	[no test files]
?   	k8s.io/heapster/events/sources/kubernetes	[no test files]
?   	k8s.io/heapster/events/util	[no test files]
ok  	k8s.io/heapster/integration	1.063s
?   	k8s.io/heapster/metrics	[no test files]
ok  	k8s.io/heapster/metrics/api/v1	1.048s
?   	k8s.io/heapster/metrics/api/v1/types	[no test files]
?   	k8s.io/heapster/metrics/core	[no test files]
ok  	k8s.io/heapster/metrics/manager	15.026s
ok  	k8s.io/heapster/metrics/processors	1.049s
ok  	k8s.io/heapster/metrics/sinks	19.064s
?   	k8s.io/heapster/metrics/sinks/gcm	[no test files]
ok  	k8s.io/heapster/metrics/sinks/hawkular	1.482s
ok  	k8s.io/heapster/metrics/sinks/influxdb	1.027s
ok  	k8s.io/heapster/metrics/sinks/kafka	1.029s
ok  	k8s.io/heapster/metrics/sinks/log	1.027s
ok  	k8s.io/heapster/metrics/sinks/metric	1.019s
ok  	k8s.io/heapster/metrics/sinks/monasca	1.396s
ok  	k8s.io/heapster/metrics/sinks/opentsdb	1.073s
ok  	k8s.io/heapster/metrics/sinks/riemann	1.023s
ok  	k8s.io/heapster/metrics/sources	8.065s
ok  	k8s.io/heapster/metrics/sources/kubelet	1.065s
ok  	k8s.io/heapster/metrics/sources/summary	1.049s
?   	k8s.io/heapster/metrics/util	[no test files]
?   	k8s.io/heapster/metrics/util/metrics	[no test files]
?   	k8s.io/heapster/version	[no test files]
godep go test -v --timeout=30m ./integration/... --vmodule=*=2  --namespace=heapster-e2e-tests --kube_versions=""
godep: WARNING: Go version (go1.6) & $GO15VENDOREXPERIMENT= wants to enable the vendor experiment, but disabling because a Godep workspace (Godeps/_workspace) exists
=== RUN   TestHeapster
--- FAIL: TestHeapster (30.00s)
	assertions.go:154: 
                        
	Location:	heapster_api_test.go:719
		
	Error:		No error is expected but got Get https://104.197.95.225/api/v1/nodes: dial tcp 104.197.95.225:443: i/o timeout
		
FAIL
exit status 1
FAIL	k8s.io/heapster/integration	30.046s
godep: go exit status 1
make: *** [test-integration] Error 1
```

--- a/deploy/docker/canary/Dockerfile
+++ b/deploy/docker/canary/Dockerfile
@@ -1,13 +1,9 @@
-FROM golang:1.4.3
+FROM golang:1.6.0
 MAINTAINER vishnuk@google.com
 
 RUN apt-get install -y git
 RUN git clone https://github.com/kubernetes/heapster.git /go/src/k8s.io/heapster
 
-RUN cd /go/src/k8s.io/heapster && make && mv heapster /heapster 
-
-# TODO: Replace the above with:
-# RUN cd /go/src/k8s.io/heapster && git checkout -b heapster-scalability remotes/origin/heapster-scalability \
-#  && make && mv heapster /heapster && mv eventer /eventer
+RUN cd /go/src/k8s.io/heapster && make && mv heapster /heapster && mv eventer /eventer
 
 ENTRYPOINT ["/heapster"]

